### PR TITLE
added top uploaders list on metrics view of list page

### PR DESCRIPTION
**Pull request by prakhar-s**
added top uploaders list on metrics view of list page,changed the order of components on this page and changed heading of "Life lst" table to "Taxa observed".

### CODE REVIEW

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import { stringify } from "querystring";
+
+export default function UploadersTable({ data, title, loadMoreUniqueSpecies, filter }) {
+  const { t } = useTranslation();
+
+  const queryParams = { ...filter };
+  if (filter.user) {
+    delete queryParams["user"];
+  }
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table-uploaders.tsx]: please avoid mutations here Instead of this you can just destructure by excluding single value like below

```jsx
const { user: _user, ...queryParams } = filter;
const address = stringify({ ...queryParams, view: "list" });
```

or even easier single liner
```jsx
const address = stringify({
  ...filter,
  view: "list",
  user: undefined,
});

```

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                    <Avatar position="relative" src={getUserImage(pic, 50)} name={name} />
+                    <ExternalBlueLink
+                      href={
+                        filter?.groupName
+                          ? `/group/${filter.groupName}/user/show/${authorId}`
+                          : `/user/show/${authorId}`
+                      }
+                    >
+                      {name}
+                    </ExternalBlueLink>
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table-uploaders.tsx]: as we discussed earlier there's better way of doing this by using `LocalLink` component like below

```jsx
<LocalLink href={`/user/show/${authorId}`}>
  <ExternalBlueLink>{name}</ExternalBlueLink>
</LocalLink>
```

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+              <Th>{t("LIST.TOP_UPLOADERS_LIST.AUTHOR_HEADER")}</Th>
+              <Th>{t("LIST.TOP_UPLOADERS_LIST.COUNT_HEADER")}</Th>
+            </Tr>
+          </Thead>
+
+          <Tbody>
+            {data.list.map(({ name, pic, authorId, count }) => (
+              <Tr key={`${authorId}-${count}`}>
+                <Td>
+                  <HStack spacing="10px">
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table-uploaders.tsx]: instead of using `10px` please use pre-defined breakpoings like `sm`, `md`, `lg` etc.

File: src/components/pages/observation/list/views/stats/table.tsx [hunk 4]
```diff
@@ -20,13 +21,21 @@ const Table = styled.table`
...
-export default function LifeListTable({ data, title, loadMoreUniqueSpecies }) {
+export default function LifeListTable({ data, title, loadMoreUniqueSpecies, filter }) {
   const { t } = useTranslation();
+
+  const queryParams = { ...filter };
+  if (filter.recoName) {
+    delete queryParams["recoName"];
+  }
+  queryParams.view = "list";
+  const address = stringify(queryParams);
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table.tsx]: please apply same suggestion here from src/components/pages/observation/list/views/stats/table-uploaders.tsx

File: src/components/pages/observation/list/views/stats/table.tsx [hunk 5]
```diff
@@ -37,7 +46,13 @@ export default function LifeListTable({ data, title, loadMoreUniqueSpecies }) {
...
                 <td>{specieName}</td>
 
-                <ExternalBlueLink href={`/observation/list?recoName=${specieName}`}>
+                <ExternalBlueLink
+                  href={
+                    filter?.groupName
+                      ? `/group/${filter.groupName}/observation/list?${address}&recoName=${specieName}`
+                      : `/observation/list?${address}&recoName=${specieName}`
+                  }
+                >
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table.tsx]: please use `<LocalLink/>` here

File: src/components/pages/observation/list/views/stats/top-uploaders.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  return (
+    <div>
+      <UploadersTable
+        data={uploader.uploadersData.data}
+        title={t("LIST.TOP_UPLOADERS_LIST.HEADING")}
+        loadMoreUniqueSpecies={uploader.uploadersData.loadMore}
+        filter={filter}
+      />
+    </div>
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/top-uploaders.tsx]: please remove div if unnecessary

**Review by harshzalavadiya — CHANGES_REQUESTED** [review]

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx (hunk 1 above)
Comment (prakhar-s) [reply src/components/pages/observation/list/views/stats/table-uploaders.tsx]: used the first approach provided

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx (hunk 2 above)
Comment (prakhar-s) [reply src/components/pages/observation/list/views/stats/table-uploaders.tsx]: used LocalLink wih params and prefixGroup parameters set

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx (hunk 3 above)
Comment (prakhar-s) [reply src/components/pages/observation/list/views/stats/table-uploaders.tsx]: used spacing={5}

File: src/components/pages/observation/list/views/stats/table.tsx (hunk 4 above)
Comment (prakhar-s) [reply src/components/pages/observation/list/views/stats/table.tsx]: done

File: src/components/pages/observation/list/views/stats/table.tsx (hunk 5 above)
Comment (prakhar-s) [reply src/components/pages/observation/list/views/stats/table.tsx]: done

File: src/components/pages/observation/list/views/stats/top-uploaders.tsx (hunk 6 above)
Comment (prakhar-s) [reply src/components/pages/observation/list/views/stats/top-uploaders.tsx]: removed the unnecessary div

File: src/components/pages/observation/list/views/stats/table.tsx [hunk 7]
```diff
@@ -46,15 +41,15 @@ export default function LifeListTable({ data, title, loadMoreUniqueSpecies, filt
...
 
-                <ExternalBlueLink
-                  href={
-                    filter?.groupName
-                      ? `/group/${filter.groupName}/observation/list?${address}&recoName=${specieName}`
-                      : `/observation/list?${address}&recoName=${specieName}`
-                  }
+                <LocalLink
+                  href={`/observation/list`}
+                  params={{ ...queryParams, view: "list", recoName: specieName }}
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table.tsx]: same here as well if you are going to override `recoName` anyway then no need to explicitly remove it above and then add it here

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx [hunk 8]
```diff
@@ -36,30 +40,22 @@ export default function UploadersTable({ data, title, loadMoreUniqueSpecies, fil
...
                 <Td>
-                  <ExternalBlueLink
-                    href={
-                      filter?.groupName
-                        ? `/group/${filter?.groupName}/observation/list?${address}&user=${authorId}`
-                        : `/observation/list?${address}&user=${authorId}`
-                    }
+                  <LocalLink
+                    href={`/observation/list`}
+                    params={{ ...queryParams, view: "list", user: authorId }}
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table-uploaders.tsx]: if you are going to override `user` anyway then there's no need to destructure it above just override it here

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx [hunk 9]
```diff
@@ -36,30 +40,22 @@ export default function UploadersTable({ data, title, loadMoreUniqueSpecies, fil
             {data.list.map(({ name, pic, authorId, count }) => (
               <Tr key={`${authorId}-${count}`}>
                 <Td>
-                  <HStack spacing="10px">
+                  <HStack spacing={5}>
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table-uploaders.tsx]: use "sm" or "md" please

File: src/components/pages/observation/list/views/stats/index.tsx [hunk 10]
```diff
@@ -1,7 +1,6 @@
 import { SimpleGrid } from "@chakra-ui/react";
 import useObservationFilter from "@components/pages/observation/common/use-observation-filter";
 import React from "react";
-
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/index.tsx:4]: Unnecessary diff

File: src/components/pages/observation/list/views/stats/table.tsx [hunk 11]
```diff
@@ -46,15 +41,15 @@ export default function LifeListTable({ data, title, loadMoreUniqueSpecies, filt
               <tr key={`${specieName}`} className="fade">
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table.tsx]: `<tr key={`${specieName}`} className="fade">`
can you refactor this as well, since key doesn't need to be in template literals this can be just be key={speciesName}

File: src/components/pages/observation/list/views/stats/table.tsx [hunk 12]
```diff
@@ -46,15 +41,15 @@ export default function LifeListTable({ data, title, loadMoreUniqueSpecies, filt
...
-                  }
+                <LocalLink
+                  href={`/observation/list`}
+                  params={{ ...queryParams, view: "list", recoName: specieName }}
+                  prefixGroup={true}
                 >
-                  <td>{specieCount}</td>
-                </ExternalBlueLink>
+                  <ExternalBlueLink>
+                    <td>{specieCount}</td>
```
Comment (harshzalavadiya) [inline src/components/pages/observation/list/views/stats/table.tsx]: `<LocalLink/>` and `<ExternalBlueLink/>` should be inside `<td/>` not outside please change that

**Review by harshzalavadiya — CHANGES_REQUESTED** [review]

File: src/components/pages/observation/list/views/stats/table-uploaders.tsx (hunk 9 above)
Comment (prakhar-s) [reply src/components/pages/observation/list/views/stats/table-uploaders.tsx]: done

**Review by harshzalavadiya — APPROVED** [review]

**Comment by harshzalavadiya** [comment]
LGTM 👍
